store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -2472,41 +2474,46 @@ public void call(String value) {
         }
     }
 
-    public void deleteMailbox(DeleteBlobs deleteBlobs) throws ServiceException {
-        StoreManager sm = StoreManager.getInstance();
-        boolean deleteStore = deleteBlobs == DeleteBlobs.ALWAYS
-                        || (deleteBlobs == DeleteBlobs.UNLESS_CENTRALIZED && !sm.supports(StoreFeature.CENTRALIZED));
-        SpoolingCache<MailboxBlob.MailboxBlobInfo> blobs = null;
+    public Set<Short> getVolumeIds() {
+        return VolumeManager.getInstance().getAllVolumes().stream().map(Volume::getId).collect(Collectors.toSet());
+    }
 
+    public void deleteMailbox(DeleteBlobs deleteBlobs) throws ServiceException {
         lock.lock();
+        MailboxMaintenance maint = null;
         try {
             // first, throw the mailbox into maintenance mode
-            //   (so anyone else with a cached reference to the Mailbox can't use it)
-            MailboxMaintenance maint = null;
-            try {
-                maint = MailboxManager.getInstance().beginMaintenance(mData.accountId, mId);
-            } catch (MailServiceException e) {
-                // Ignore wrong mailbox exception.  It may be thrown if we're redoing a DeleteMailbox that was interrupted
-                // when server crashed in the middle of the operation.  Database says the mailbox has been deleted, but
-                // there may be other files that still need to be cleaned up.
-                if (!MailServiceException.WRONG_MAILBOX.equals(e.getCode())) {
-                    throw e;
-                }
-            }
-
+            // (so anyone else with a cached reference to the Mailbox can't use it)
+            maint = MailboxManager.getInstance().beginMaintenance(mData.accountId, mId);
             DeleteMailbox redoRecorder = new DeleteMailbox(mId);
+            beginTransaction("deleteMailbox", null, redoRecorder);
             boolean needRedo = needRedo(null, redoRecorder);
-            boolean success = false;
+            boolean success = false, deleteStore = false;
+            Set<Short> volumeIds = getVolumeIds();
+            for (short id : volumeIds) {
+                StoreManager sm = StoreManager.getReaderSMInstance(id);
+                //if sm is of ExternalStoreManager deleteStore will always be true
+                deleteStore = (deleteBlobs == DeleteBlobs.ALWAYS
+                        || (deleteBlobs == DeleteBlobs.UNLESS_CENTRALIZED && sm.checkIfStoreManagerIsExternal()));
+                SpoolingCache<MailboxBlob.MailboxBlobInfo> blobs = null;
+                if (deleteStore && !sm.supports(StoreManager.StoreFeature.BULK_DELETE)) {
+                    blobs = DbMailItem.getAllBlobs(this);
+                }
+                if (deleteStore) {
+                    try {
+                        sm.deleteStore(this, blobs);
```

Comment:
@rykoon-amateur I hope this removed code for fileBlobStore delete `if (deleteStore && !sm.supports(StoreManager.StoreFeature.BULK_DELETE)) {
                    blobs = DbMailItem.getAllBlobs(this);
                }`
                
is handled in that class.
Please confirm